swagger.yaml

```diff
@@ -1,42 +1,254 @@
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  NewJob:
     type: object
+    required:
+      - "last_updated"
+      - "search_index_name"
+      - "state"
     properties:
-      message:
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
         type: string
-        description: "Message returned by hello world endpoint"
-        example: "Hello, world!"
\ No newline at end of file
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  Jobs:
+    type: object
+    required:
+      - "count"
+      - "items"
+      - "limit"
+      - "offset"
+      - "total_count"
+    properties:
+      count:
+        type: integer
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Job'
+      limit:
+        type: integer
+      offset:
+        type: integer
+      total_count:
+        type: integer
+      sort:
+        type: integer
+
+parameters:
+
+  id:
+    name: id
+    description: The id of a search reindex job when job POSTed (Job.id)
+    in: path
+    required: true
+    type: string
+
+  limit:
+    name: limit
+    description: The maximum number of Job objects we're returning in this Jobs object. The default limit is 20.
+    in: query
+    type: integer
+    default: 20
+
+  offset:
+    name: offset
+    description: The number of Job objects into the full list that this particular response is starting at. The default number to start at is 0.
+    in: query
+    type: integer
+    default: 0
+
+  sort:
+    name: sort
+    description: Sort on last_updated field
```

Comment:
The default for sort parameter will be on the `last_updated` field, so can add default parameter to sort.

The sort query parameter could be set to something other than `last_updated` and hence this must be clear in the description, something like:

`description: The sort order in which to return a list of reindex job resources. This defaults to last_updated in ascending order.`

Reply:
The sort field is confusing me a little. At first I took it to be a boolean i.e. true means sort on the last_updated field, false means don't sort. However, I noticed that the spec has it down as an integer. In that case I'm not sure what its default value should be. However, the way you have described it here makes it sound like a string with a default value of 'last_updated' so that's what I've done. Is that correct?

Reply:
Nathan has confirmed that sort is a string